tests/ops/test_qubit_ops.py

```diff
@@ -330,6 +330,76 @@ def test_matrices(self, ops, mat, tol):
         res = op.matrix
         assert np.allclose(res, mat, atol=tol, rtol=0)
 
+    def test_x_decomposition(self, tol):
+        """Tests that the decomposition of the PauliX is correct"""
+        op = qml.PauliX(wires=0)
+        res = op.decomposition(0)
+
+        assert len(res) == 1
+
+        assert res[0].name == "RX"
+        assert res[0].wires == [0]
+        assert res[0].params[0] == np.pi
+
+    def test_y_decomposition(self, tol):
+        """Tests that the decomposition of the PauliX is correct"""
+        op = qml.PauliY(wires=0)
+        res = op.decomposition(0)
+
+        assert len(res) == 1
+
+        assert res[0].name == "RY"
+        assert res[0].wires == [0]
+        assert res[0].params[0] == np.pi
+
+    def test_z_decomposition(self, tol):
+        """Tests that the decomposition of the PauliX is correct"""
+        op = qml.PauliZ(wires=0)
+        res = op.decomposition(0)
+
+        assert len(res) == 1
+
+        assert res[0].name == "RZ"
+        assert res[0].wires == [0]
+        assert res[0].params[0] == np.pi
+
+    def test_s_decomposition(self, tol):
+        """Tests that the decomposition of the PauliX is correct"""
+        op = qml.S(wires=0)
+        res = op.decomposition(0)
+
+        assert len(res) == 1
+
+        assert res[0].name == "RZ"
+        assert res[0].wires == [0]
+        assert res[0].params[0] == np.pi / 2
+
+    def test_t_decomposition(self, tol):
+        """Tests that the decomposition of the PauliX is correct"""
+        op = qml.T(wires=0)
+        res = op.decomposition(0)
+
+        assert len(res) == 1
+
+        assert res[0].name == "RZ"
+        assert res[0].wires == [0]
+        assert res[0].params[0] == np.pi / 4
+
+    def test_hadamard_decomposition(self, tol):
+        """Tests that the decomposition of the PauliX is correct"""
+        op = qml.Hadamard(wires=0)
+        res = op.decomposition(0)
+
+        assert len(res) == 2
+
+        assert res[0].name == "RX"
+        assert res[0].wires == [0]
+        assert res[0].params[0] == np.pi / 2
+
+        assert res[1].name == "RZ"
+        assert res[1].wires == [0]
+        assert res[0].params[0] == np.pi / 2
```

Comment:
Really thorough testing @Lucaman99! I would recommend however including numeric checks to ensure that the decomposition is correct (~up to global phase). For example,
```python
    def test_hadamard_decomposition(self, tol):
        """Tests that the decomposition of the PauliX is correct"""
        op = qml.Hadamard(wires=0)
        res = op.decomposition(0)

        decomposed_matrix = np.linalg.multi_dot([i.matrix for i in reversed(res)])
        global_phase = (decomposed_matrix[op.matrix != 0] / op.matrix[op.matrix != 0])[0]

        assert np.allclose(decomposed_matrix, global_phase * op.matrix, atol=tol, rtol=0)
```